demos/dox/align/local_alignment_unbanded_score_execution_policy_wavefront.cpp

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+
+#include <seqan/align_parallel.h>
+#include <seqan/stream.h>  // for printing strings
+
+int main()
+{
+    using TSequence = seqan::String<seqan::Dna>;
+    using TThreadModel = seqan::WavefrontAlignment<seqan::BlockOffsetOptimization>;
+    using TVectorSpec = seqan::Vectorial;
+    using TExecPolicy = seqan::ExecutionPolicy<TThreadModel, TVectorSpec>;
+
+    // dummy sequences
+    TSequence seqH;
+    TSequence seqV;
+
+    for (size_t i = 0; i < 10000; ++i)
+    {
+        seqan::appendValue(seqH, 'A');
+        seqan::appendValue(seqV, 'A');
+    }
+
+    seqan::StringSet<TSequence> seqs1;
+    seqan::StringSet<TSequence> seqs2;
+
+    for (size_t i = 0; i < 100; ++i)
+
+    {
+        seqan::appendValue(seqs1, seqH);
+        seqan::appendValue(seqs2, seqV);
+    }
+
+    TExecPolicy execPolicy;
+    seqan::setBlockSize(execPolicy, 500); // Sets the size of blocks used internally to partition the alignment matrix.
+    seqan::setParallelAlignments(execPolicy, 10); // Compute ten alignments at the same time.
+    seqan::setNumThreads(execPolicy, 4); // Use four threads to compute the actual alignment.
+
+    seqan::Score<int32_t, seqan::Simple> scoreAffine(2, -2, -1, -4);
+
+    seqan::String<int32_t> scores = seqan::localAlignmentScore(execPolicy, seqs1, seqs2, scoreAffine);
```

Comment:
This seems to also fail.

include/seqan/align_parallel/async_wave_execution_interface.h:241:44: error: no matching function for call to ‘seqan::impl::AsyncWaveAlignExecutor<const seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> >, const seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> >, seqan::DPSettings<seqan::Score<int, seqan::Tag<seqan::Simple_> >, seqan::impl::doWaveAlignment(const seqan::ExecutionPolicy<seqan::WavefrontAlignment<TSpec>, TVectorizationSpec>&, const TAlgorithmSpec&, const TSetH&, const TSetV&, const TScore&, TArgs&& ...) [with TWaveSpec = seqan::Tag<seqan::BlockOffsetOptimization_>; TVectorizationPolicy = seqan::Tag<seqan::Vectorial_>; TAlgorithmSpec = seqan::LocalAlignment_<>; TSetH = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> > >; TSetV = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> > >; TScore = seqan::Score<int, seqan::Tag<seqan::Simple_> >; TArgs = {}; typename std::enable_if<((! std::is_same<seqan::WavefrontAlignment<TSpec>, seqan::Tag<seqan::Serial_> >::value) && (! std::is_same<seqan::WavefrontAlignment<TSpec>, seqan::Tag<seqan::Parallel_> >::value)), int>::type <anonymous> = 0]::DPConfigTraits> >::AsyncWaveAlignExecutor(const seqan::DPSettings<seqan::Score<int, seqan::Tag<seqan::Simple_> >, seqan::impl::doWaveAlignment(const seqan::ExecutionPolicy<seqan::WavefrontAlignment<TSpec>, TVectorizationSpec>&, const TAlgorithmSpec&, const TSetH&, const TSetV&, const TScore&, TArgs&& ...) [with TWaveSpec = seqan::Tag<seqan::BlockOffsetOptimization_>; TVectorizationPolicy = seqan::Tag<seqan::Vectorial_>; TAlgorithmSpec = seqan::LocalAlignment_<>; TSetH = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> > >; TSetV = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna_>, seqan::Alloc<> > >; TScore = seqan::Score<int, seqan::Tag<seqan::Simple_> >; TArgs = {}; typename std::enable_if<((! std::is_same<seqan::WavefrontAlignment<TSpec>, seqan::Tag<seqan::Serial_> >::value) && (! std::is_same<seqan::WavefrontAlignment<TSpec>, seqan::Tag<seqan::Parallel_> >::value)), int>::type <anonymous> = 0]::DPConfigTraits>&, const seqan::ExecutionPolicy<seqan::WavefrontAlignment<seqan::Tag<seqan::BlockOffsetOptimization_> >, seqan::Tag<seqan::Vectorial_> >&)’